athenet/sparsifying/utils/interval.py

```diff
@@ -0,0 +1,284 @@
+        return Interval(self.lower[at], self.upper[at])
+
+    def __setitem__(self, at, other):
+        """Just like Theano set_subtensor function, but as a operator.
+
+        :at: Coordinates / slice to be set.
+        :other: Data to be put at 'at'
+
+        """
+        self.lower = T.set_subtensor(self.lower[at], other.lower)
+        self.upper = T.set_subtensor(self.upper[at], other.upper)
+
+    def __repr__(self):
+        """Standard repr method."""
+        return '[' + repr(self.lower) + ', ' + repr(self.upper) + ']'
+
+    def __str__(self):
+        """"Standard str method."""
+        return '[' + str(self.lower) + ', ' + str(self.upper) + ']'
+
+    def shape(self):
+        """Returns shape of interval. Checks only 'lower' matrix.
+
+        Note: does not require self.upper for computations. Therefore it is not
+        safe, but faster."""
+        return self.lower.shape
+
+    def __add__(self, other):
+        """Returns sum of two intervals.
+
+        :other: Interval to be added."""
+        if isinstance(other, Interval):
+            res_lower = self.lower + other.lower
+            res_upper = self.upper + other.upper
+        else:
+            res_lower = self.lower + other
+            res_upper = self.upper + other
+        return Interval(res_lower, res_upper)
+
+    __radd__ = __add__
+
+    def __sub__(self, other):
+        """Returns difference between two intervals.
+
+        :other: Interval to be subtracted."""
+        if isinstance(other, Interval):
+            res_lower = self.lower - other.upper
+            res_upper = self.upper - other.lower
+        else:
+            res_lower = self.lower - other
+            res_upper = self.upper - other
+        return Interval(res_lower, res_upper)
+
+    def __rsub__(self, other):
+        """Returns diffeerence between number and interval.
+
+        :other: A number that self will be subtracted from."""
+        res_lower = other - self.upper
+        res_upper = other - self.lower
+        return Interval(res_lower, res_upper)
+
+    def __mul__(self, other):
+        """Returns product of two intervals.
+
+        :other: Interval to be multiplied."""
+        if isinstance(other, Interval):
+            ll = self.lower * other.lower
+            lu = self.lower * other.upper
+            ul = self.upper * other.lower
+            uu = self.upper * other.upper
+            l = T.minimum(ll, lu)
+            l = T.minimum(l, ul)
+            l = T.minimum(l, uu)
+            u = T.maximum(ll, lu)
+            u = T.maximum(u, ul)
+            u = T.maximum(u, uu)
+            return Interval(l, u)
+        else:
+            ll = self.lower * other
+            uu = self.upper * other
+            l = T.minimum(ll, uu)
+            u = T.maximum(ll, uu)
+            return Interval(l, u)
+
+    __rmul__ = __mul__
+
+    def __div__(self, other):
+        """Returns quotient of self and other.
+
+        :other: Divisor.
```

Comment:
I need you to say more (is it Interval, common number or mushroom? - this information is necessary)